staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -224,6 +224,10 @@ func (a *Allocator) Allocate(ctx context.Context, node *v1.Node) (finalResult []
 						}
 					}
 				}
+				// If we don't have any devices, then we cannot allocate here as "all" means "at least one" in this case.
```

Comment:
This PR should also clarify that in the API description.

Reply:
Thank you for the review! Updated `AllocationMode` description, PTAL.